FWCore/Framework/src/ProductResolvers.h

```diff
@@ -307,12 +308,14 @@ namespace edm {
     // for "product" view
     ProductData productData_;
     Worker* worker_ = nullptr;
-    mutable WaitingTaskList waitingTasks_;
+    CMS_THREAD_SAFE mutable WaitingTaskList waitingTasks_;
     mutable std::atomic<bool> prefetchRequested_;
     // for provenance
     ParentageID parentageID_;
     // for filter in a Path
-    mutable ProductStatus status_;
+    // The variable is only modified or read at times where the
+    //  framework has guaranteed synchronization between write and read
+    CMS_THREAD_SAFE mutable ProductStatus status_;
```

Comment:
Following our chat, this one likely should be `std::atomic<ProductStatus>` (as `DataManagingOrAliasProductResolver::theStatus_` is, but I'll take a closer look (=remind myself) how exactly it is being used (e.g. if there is a need for synchronization between threads).

Reply:
We will address this issue in another pull request.